include/swift/AST/DiagnosticsSema.def

```diff
@@ -464,6 +464,8 @@ ERROR(cannot_convert_assign_protocol,none,
       (Type, Type))
 ERROR(cannot_convert_assign_nil,none,
       "'nil' cannot be assigned to type %0", (Type))
+NOTE(assign_protocol_conformance,none,
+     "did you mean to add conformance to %0?", (Type))
```

Comment:
It might be nice to restate both the type and protocol here, but I'll leave that up to you.

Reply:
I did that initially, but then removed it because the note is attached to the decl.

Reply:
Actually, it should be attached to the anchor, so I am reverting to what I had before.